design/priority-class-name-support_design.md

```diff
@@ -0,0 +1,587 @@
+   - Example: `velero-low` with value 10
+   - Rationale: These are temporary workloads that can be delayed during resource contention
+
+This hierarchy ensures that core Velero components remain operational even under resource pressure, while allowing less critical workloads to be preempted if necessary.
+
+This approach has several advantages:
+
+- Leverages existing configuration mechanisms, minimizing new CLI flags
+- Provides a single point of configuration for related components (node agent and its pods)
+- Allows dynamic configuration updates without requiring Velero reinstallation
+- Maintains backward compatibility with existing installations
+- Enables administrators to set up priority classes during initial deployment
+- Keeps configuration simple by using the same priority class for all maintenance jobs
+
+The priority class name for data mover pods will be determined by checking the node-agent-configmap. This approach provides a centralized way to configure priority class names for all data mover pods. The same approach will be used for PVB (PodVolumeBackup) and PVR (PodVolumeRestore) pods, which will also retrieve their priority class name from the node-agent-configmap.
+
+For PVB and PVR pods specifically, the controllers will need to be updated to retrieve the priority class name from the node-agent-configmap and pass it to the pod creation functions. For example, in the PodVolumeBackup controller:
+
+```go
+// In pkg/controller/pod_volume_backup_controller.go
+priorityClassName, _ := kube.GetDataMoverPriorityClassName(ctx, namespace, kubeClient, configMapName)
+
+// Add priorityClassName to the pod spec
+pod := &corev1api.Pod{
+    // ... existing fields ...
+    Spec: corev1api.PodSpec{
+        // ... existing fields ...
+        PriorityClassName: priorityClassName,
+    },
+}
+```
+
+Similarly, in the PodVolumeRestore controller:
+
+```go
+// In pkg/controller/pod_volume_restore_controller.go
+priorityClassName, _ := kube.GetDataMoverPriorityClassName(ctx, namespace, kubeClient, configMapName)
+
+// Add priorityClassName to the pod spec
+pod := &corev1api.Pod{
+    // ... existing fields ...
+    Spec: corev1api.PodSpec{
+        // ... existing fields ...
+        PriorityClassName: priorityClassName,
+    },
+}
+```
+
+### VGDP Micro-Service Considerations
+
+With the introduction of VGDP micro-services (as described in the VGDP micro-service design), data mover pods are created as dedicated pods for volume snapshot data movement. These pods will also inherit the priority class configuration from the node-agent-configmap. Since VGDP-MS pods (backupPod/restorePod) inherit their configurations from the node-agent, they will automatically use the priority class name specified in the node-agent-configmap.
+
+This ensures that all pods created by Velero for data movement operations (including VGDP micro-service pods, PVB, and PVR) use a consistent approach for priority class name configuration through the node-agent-configmap.
```

Comment:
PVB/PVR is also using VGDP, so here we better say:
`or data movement operations (CSI snapshot data movement, PVB, and PVR)`